tools/uvmboot/main.go

```diff
@@ -0,0 +1,264 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/Microsoft/hcsshim/internal/uvm"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+const (
+	kernelArgsArgName           = "kernel-args"
+	rootFSTypeArgName           = "root-fs-type"
+	vpMemMaxCountArgName        = "vpmem-max-count"
+	vpMemMaxSizeArgName         = "vpmem-max-size"
+	cpusArgName                 = "cpus"
+	memoryArgName               = "memory"
+	allowOvercommitArgName      = "allow-overcommit"
+	enableDeferredCommitArgName = "enable-deferred-commit"
+	measureArgName              = "measure"
+	parallelArgName             = "parallel"
+	countArgName                = "count"
+	kernelDirectArgName         = "kernel-direct"
+	execCommandLineArgName      = "exec"
+	forwardStdoutArgName        = "fwd-stdout"
+	forwardStderrArgName        = "fwd-stderr"
+	debugArgName                = "debug"
+	guestConnectionArgName      = "guest-con"
+	outputHandlingArgName       = "output-handling"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "uvmboot"
+	app.Usage = "Boot a utility VM"
+
+	app.Flags = []cli.Flag{
+		cli.Uint64Flag{
+			Name:  cpusArgName,
+			Usage: "Number of CPUs on the UVM. Uses hcsshim default if not specified",
+		},
+		cli.UintFlag{
+			Name:  memoryArgName,
+			Usage: "Amount of memory on the UVM, in MB. Uses hcsshim default if not specified",
+		},
+		cli.BoolFlag{
+			Name:  measureArgName,
+			Usage: "Measure wall clock time of the UVM run",
+		},
+		cli.IntFlag{
+			Name:  parallelArgName,
+			Value: 1,
+			Usage: "Number of UVMs to boot in parallel",
+		},
+		cli.IntFlag{
+			Name:  countArgName,
+			Value: 1,
+			Usage: "Total number of UVMs to run",
+		},
+		cli.BoolFlag{
+			Name:  allowOvercommitArgName,
+			Usage: "Allow memory overcommit on the UVM",
+		},
+		cli.BoolFlag{
+			Name:  enableDeferredCommitArgName,
+			Usage: "Enable deferred commit on the UVM",
+		},
+		cli.BoolFlag{
+			Name:  debugArgName,
+			Usage: "Enable debug level logging in HCSShim",
+		},
+		cli.BoolFlag{

```

Comment:
Do you really need this as an opt? What case do we want a guest connection for this tool?

Reply:
I thought there may be a use case for being able to boot a normal UVM and have the logrus output streaming to your console, so you can view it as you launch containers in that UVM. With the current configuration, uvmboot can be used to launch a completely standard UVM as well as a "test" UVM.

Reply:
This is a really weird scenario. This would work you are correct but it would boot successfully, negotiate with the bridge, and then hang forever waiting for a create container/container exit sequence. To support this requires some more thought I think. Lets remove it for now.